sys/riscv/riscv/pmap.c

```diff
@@ -3399,19 +3472,27 @@ pmap_zero_page_area(vm_page_t m, int off, int size)
 void
 pmap_copy_page(vm_page_t msrc, vm_page_t mdst)
 {
-	vm_offset_t src = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(msrc));
-	vm_offset_t dst = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(mdst));
+	vm_ptr_t src = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(msrc));
+	vm_ptr_t dst = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(mdst));
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	src = (vm_ptr_t)cheri_setbounds((void *)src, PAGE_SIZE);
```

Comment:
Probably a no-op but should probably use setboundsexact here as well.

Reply:
I've fixed this and several others in pmap.c.  I also audited other cases of cheri_setbounds that couldn't change and found a couple of other bugs with representability that are now fixed as well.